TypeCobol/Compiler/Sql/Scanner/SqlScanner.cs

```diff
@@ -50,46 +80,248 @@ public override Token GetTokenStartingFrom(int startIndex)
+                    if (TryScanDecimalFloatingPointSpecialValue(startIndex, false,
+                            out var tokenValue))
+                    {
+                        Token decimalFloatingPointSpecialValueToken = new Token(
+                            TokenType.SQL_DecimalFloatingPointLiteral,
+                            startIndex, currentIndex - 1, tokensLine)
+                        {
+                            LiteralValue = tokenValue
+                        };
+                        return decimalFloatingPointSpecialValueToken;
+                    }
+
+                    return ScanKeywordOrUserDefinedWord(startIndex);
+                case '\'':
+                case '"':
+                    return ScanAlphanumericLiteral(startIndex, TokenType.AlphanumericLiteral, null);
+                case 'X':
+                case 'x':
+                    if (line[currentIndex + 1] == '\'' || line[currentIndex + 1] == '"')
+                    {
+                        return ScanAlphanumericLiteral(startIndex, TokenType.AlphanumericLiteral, null);
+                    }
+                    else
+                    {
+                        return ScanKeywordOrUserDefinedWord(startIndex);
+                    }
+                case 'B':
+                case 'b':
+                    if ((startIndex < lastIndex - 2) && (line[startIndex + 1] == 'x' || line[startIndex + 1] == 'X') &&
+                        (line[startIndex + 2] == '\'' || line[startIndex + 2] == '"'))
+                    {
+                        return ScanAlphanumericLiteral(startIndex, TokenType.SQL_BinaryStringLiteral, null);
+                    }
+                    else
+                    {
+                        return ScanKeywordOrUserDefinedWord(startIndex);
+                    }
+                case 'U':
+                case 'u':
+                case 'G':
+                case 'g':
+                    if ((startIndex < lastIndex - 2) && (line[startIndex + 1] == 'x' || line[startIndex + 1] == 'X') &&
+                        (line[startIndex + 2] == '\'' || line[startIndex + 2] == '"'))
+                    {
+                        return ScanAlphanumericLiteral(startIndex, TokenType.SQL_GraphicStringLiteral, null);
+                    }
+                    else
+                    {
+                        return ScanKeywordOrUserDefinedWord(startIndex);
+                    }
+                default:
+                    return ScanKeywordOrUserDefinedWord(startIndex);
 
             }
+        }
 
-            if (IsSqlKeywordPart(line[currentIndex]))
+        private Token ScanSqlNumericLiteral(int startIndex, bool isNegative)
+        {
+            var tryScanDecimalFloatingPointSpecialValue =
+                TryScanDecimalFloatingPointSpecialValue(startIndex, isNegative,
+                    out var decimalFloatingPointLiteralTokenValue);
+            if (tryScanDecimalFloatingPointSpecialValue == true)
```

Comment:
This can be simplified as

```csharp
if (TryScanDecimalFloatingPointSpecialValue(startIndex, isNegative, out var decimalFloatingPointLiteralTokenValue))
{
    [...]
}
```